lib/post.rb

```diff
@@ -0,0 +1,35 @@
+require 'pg'
+
+class Post
+  attr_reader :id, :date, :author, :message
+
+  def initialize(id:, date:, author:, message:)
+    @id = id
+    @date = date
+    @author = author
+    @message = message
+  end
+
+  def self.all
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
```

Comment:
These lines are present in all your methods that access the db - they could be extracted into a database helper class :)